rcl_executor/src/handle.c

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository https://github.com/micro-ROS/rcl_executor.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl_executor/handle.h"
+
+#include <stdio.h>
+
+#include <rcl/error_handling.h>
+#include <rcutils/logging_macros.h>
+
+// initialization of info object
+rcl_ret_t
+rcle_handle_size_zero_init(rcle_handle_size_t * info)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(info, RCL_RET_INVALID_ARGUMENT);
+
+  info->number_of_subscriptions = 0;
+  info->number_of_guard_conditions = 0;
+  info->number_of_timers = 0;
+  info->number_of_clients = 0;
+  info->number_of_events = 0;
+  info->number_of_services = 0;
```

Comment:
i would use memset to set all zero data with size, so that i do not need to change this every single time if structure changed.

Reply:
Resolved in a25dbe55bc8cc8237c5f5bfa8aa98c0bfc21d078